feature/experimental/tunnel/otg_tests/tunnel_interface_based_ipv4_gre_encapsulation_test/tunnel_interface_based_ipv4_gre_encapsulation_test.go

```diff
@@ -0,0 +1,482 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tunnel_interface_based_ipv4_gre_encapsulation_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/open-traffic-generator/snappi/gosnappi"
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/otgutils"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+)
+
+const (
+	tunnelNhIpv4Network      = "198.18.1.0"
+	tunnelSrcIpv4Network     = "198.18.2.0"
+	tunnelDesIpv4Network     = "198.18.32.0"
+	encapInnerDesIpv4Network = "198.18.100.1"
+	decapInnerDesIpv4Network = "198.18.200.1"
+	tunnelPlen4              = 31
+	plen4                    = 32
+	interfacePlen4           = 24
+	tunnelCount              = 32
+	tunnelInterface          = "fti0"
+	trafficRatePps           = 5000
+	trafficDuration          = 120
+	tolerance                = 12
+)
+
+var (
+	dutIntf1 = attrs.Attributes{
+		Desc:    "dutsrc",
+		MAC:     "00:00:a1:a1:a1:a1",
+		IPv4:    "198.18.0.1",
+		IPv4Len: interfacePlen4,
+	}
+
+	dutIntf2 = attrs.Attributes{
+		Desc:    "dutdst1",
+		MAC:     "00:00:b1:b1:b1:b1",
+		IPv4:    "198.18.3.1",
+		IPv4Len: interfacePlen4,
+	}
+
+	dutIntf3 = attrs.Attributes{
+		Desc:    "dutdst2",
+		MAC:     "00:00:c1:c1:c1:c1",
+		IPv4:    "198.18.4.1",
+		IPv4Len: interfacePlen4,
+	}
+
+	otgIntf1 = attrs.Attributes{
+		Name:    "otgsrc",
+		IPv4:    "198.18.0.2",
+		IPv4Len: interfacePlen4,
+		MAC:     "00:00:01:01:01:01",
+	}
+
+	otgIntf2 = attrs.Attributes{
+		Name:    "otgdst1",
+		IPv4:    "198.18.3.2",
+		IPv4Len: interfacePlen4,
+		MAC:     "00:00:02:02:02:02",
+	}
+
+	otgIntf3 = attrs.Attributes{
+		Name:    "otgdst2",
+		IPv4:    "198.18.4.2",
+		IPv4Len: interfacePlen4,
+		MAC:     "00:00:03:03:03:03",
+	}
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func TestTunnelEncapsulationByGREOverIPv4WithLoadBalance(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dutPort1 := dut.Port(t, "port1")
+	dutPort2 := dut.Port(t, "port2")
+	dutPort3 := dut.Port(t, "port3")
+	ate := ondatra.ATE(t, "ate")
+	ateport1 := ate.Port(t, "port1")
+	ateport2 := ate.Port(t, "port2")
+	ateport3 := ate.Port(t, "port3")
+	egressInterfaces := []string{dutPort2.Name(), dutPort3.Name()}
+	initialEgressPkts := make([]uint64, tunnelCount)
+	initialTunnelInPkts := make([]uint64, tunnelCount)
+	initialTunnelOutPkts := make([]uint64, tunnelCount)
+	tunnelLoadblanceDiff := tunnelCount * 3
+	interfaceLoadblanceDiff := tolerance
+	t.Run("Configure dut with 32 tunnel interface with one ingress and 2 egress interface", func(t *testing.T) {
```

Comment:
t.Run() will create sub-testcase. There should be no dependency requirement between sub-testcases as you can run any sub-testcase independently. But, your sub-testcases are required to run one by one in fixed order.